src/backend/distributed/planner/distributed_planner.c

```diff
@@ -556,6 +558,12 @@ CreateDistributedPlannedStmt(uint64 planId, PlannedStmt *localPlan, Query *origi
 		RaiseDeferredError(distributedPlan->planningError, ERROR);
 	}
 
+	/* traverse distributedPlan and extract the list of used subplans */
+	if (ShouldFindSubPlansUsedInPlan(distributedPlan))
+	{
+		FindSubPlansUsedInPlan(distributedPlan);
```

Comment:
What happens if we use the originalQuery instead of `distributedPlan == ->jobQuery`, I don't want to rely on `jobQuery` field, which is very much dependent how the distributed planning happens.


Do you see anything that prevents us using `originalQuery` to find the intermediate results?

Reply:
We lose some of the information, and end up not sending the intermediate results to some of the workers.

You can see one failed attempt here: https://github.com/citusdata/citus/tree/hanefi-failing-originalquery